great_expectations/core/usage_statistics/usage_statistics.py

```diff
@@ -198,7 +198,10 @@ def build_envelope(self, message: dict) -> dict:
     @staticmethod
     def validate_message(message: dict, schema: dict) -> bool:
         try:
-            jsonschema.validate(message, schema=schema)
+            jsonschema.validators.Draft202012Validator(
+                schema=schema,
+                format_checker=jsonschema.validators.Draft202012Validator.FORMAT_CHECKER,
```

Comment:
`format_checker=jsonschema.validators.Draft202012Validator.FORMAT_CHECKER`

i shamefully don't really understand how the whole `jsonschema` thing works. where is this defined? is it visible in the repo? you import the `jsonschema` module, which is presumably installed with pip, and somehow it has the custom validations you defined?

Reply:
what happens to the message if validation fails? it gets logged but the message doesn't get written? so for a UUID value that fails we get an empty string instead of str(UUID)?

Reply:
This is all handled in the AWS lambda that accepts the events. The schema that we generate by running `schemas.py` (as a json file) is included in the lambda deployment and used by the lambda to validate events. Take a look at `https://github.com/greatexpectationslabs/usage_statistics`, which defines the lambda and uses the `serverless-reqvalidator-plugin` to actually do the validation: https://github.com/greatexpectationslabs/usage_statistics/blob/main/usage_statistics/serverless.yml#L4 Happy to do a walkthrough / answer questions of what I know.

Reply:
I believe it rejects the event but I would verify that. Here is a link to docs on the plugin: https://www.serverless.com/plugins/serverless-reqvalidator-plugin